src/common/transformations/src/transformations/resolve_names_collisions.cpp

```diff
@@ -20,63 +35,28 @@ bool ov::pass::ResolveNameCollisions::run_on_model(const std::shared_ptr<ov::Mod
     std::vector<Node*> nodes_with_conflicts;
     std::unordered_map<std::string, std::list<Node*>> visited_nodes;
 
-    for (const auto& node : model->get_ordered_ops()) {
-        // Detect names collisions only for nodes with autogenerated names
-        const auto& friendly_name = node->get_friendly_name();
-        visited_nodes[friendly_name].emplace_back(node.get());
-    }
+    collect_name_collisions_map(model, visited_nodes);
 
-    for (const auto& l_nodes : visited_nodes) {
-        if (l_nodes.second.size() == 1)
+    for (const auto& it : visited_nodes) {
+        const auto& same_named_ops = it.second;
+        if (same_named_ops.size() < 2) {
             continue;
-        const size_t nodes_size = l_nodes.second.size();
-        bool has_public_node = false;  // Parameter, Result ans Sinks
-        size_t i(0);
-        for (auto* node : l_nodes.second) {
-            i++;
-            // Skip the last node if we don't have public nodes with collisions
-            if (i == nodes_size && !has_public_node)
-                break;
-            if (dynamic_cast<const ov::op::v0::Result*>(node)) {
-                // Result is a service node
+        }
+
+        int64_t cnt = 2;
+        for (const auto& op : same_named_ops) {
+            // check if op has OV autogenerated friendly name. unique and friendly names have to be equal.
+            bool is_autogenerated = op->m_friendly_name.empty();
+            if (!is_autogenerated) {
                 continue;
             }
-            if (dynamic_cast<const ov::op::Sink*>(node) || dynamic_cast<const ov::op::v0::Parameter*>(node)) {
-                // Resolve names for public ops with autogenerated name
-                if (node->m_friendly_name.empty())
-                    nodes_with_conflicts.emplace_back(node);
-                has_public_node = true;
-                continue;
-            } else {
-                // For result we need to avoid changes in previous operation
-                bool is_public = false;
-                for (const auto& output : node->outputs()) {
-                    for (const auto input : output.get_target_inputs()) {
-                        if (dynamic_cast<const ov::op::v0::Result*>(input.get_node())) {
-                            has_public_node = true;
-                            is_public = true;
-                            break;
-                        }
-                    }
-                    if (is_public)
-                        break;
-                }
-                if (is_public)
-                    continue;
+            // add a prefix "_counter" to the autogenerated name to make it unique.
+            auto new_name = op->get_friendly_name() + "_" + std::to_string(cnt++);
```

Comment:
what is the difference between `m_friendly_name` adn `get_fritendly_name()`?

Reply:
Don't we get a name of a form `_3`?

Reply:
get_fritendly_name() will return autogenerated name, if m_friendly_name is empty
m_friendly_name will remain empty

Reply:
we will get _2
after that cnt will be incremented